hips/hip-9999.md

```diff
@@ -0,0 +1,317 @@
+- Message serialization/RPC (GRPC/protobuf, JSON, etc)
+Rather than trying to implement a rich FFI, either directly via Wasip1, or indirectly via WIT. Implement/utilize a simple FFI that uses a higher level serialization ie. serialization of internal types into e.g. protobufs or JSON and utilizing e.g. a simple FFI or GRPC or HTTP for the majority of message content.
+
+## Rejected ideas
+
+*Why certain ideas that were brought while discussing this HIP were not ultimately pursued.*
+
+WIP
+
+### 🚫 A central repository for plugins
+
+- A centralized repository for community plugins. Helm moved from a central charts repository to a distributed one [for documented reasons](https://github.com/helm/community/blob/main/hips/archives/helm/distributed-search.md) that we can anticipate applying here—including allowing scaling as the number of charts grew, and ecosystem growth through community-managed alternatives.
+- Go native plugin system https://pkg.go.dev/plugin. It does not support Windows
+
+### 🚫 Non-WASM plugins
+
+1. OS/Arch distribtuion problems
+2. Security concerns of non-sandboxed execution
+
+These include:
+- Go native plugin system
+https://pkg.go.dev/plugin
+    - downsides:
+        - does not support Windows. For this reason alone, we can't use it
+        - also the version of go, architecture, and (some other build options) MUST match between parent and plugin
+        - this means WHEN helm core go version upgrades, none of the community plugins will work until the plugin maintainers upgrade their plugin to match the new helm core Go version
+- Hashcorp's go-plugin package
+https://github.com/hashicorp/go-plugin
+    - this is also what protobuf does (subprocess)
+- embedded go interpreter
+    - traefik does this (github.com/traefik/yaegi)
+- Caddy novel Go plugin approach
+https://caddyserver.com/docs/architecture
+
+### 🚫 Non-Extism WASM plugins
+
+- wazero (directly)
+https://github.com/tetratelabs/wazero
+    - note that [extism/go-sdk](https://github.com/extism/go-sdk) (see Extism under [Reference implementation](#reference-implementation)) leverages wazero in a viable way.
+    - downsides:
+        - all params are uint32. So any strings must be interpreted from that
+or in our case structs must be converted in/out (may need to write a library for this?)
+        - The allocation example uses CGO to allocate memory, uses unsafe package. note we don't need to use tinygo
+- Wadge in wasmCloud
+https://wasmcloud.com/docs/developer/languages/go/components#wadge
+    - Bailey Cosmonic CTO suggested (via Taylor Thomas)
+    - downsides:
+        - relies on CGO
+- Envoy's Wasm approach
+https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/wasm_filter
+- WIT/WASI Preview 2 (WASM IDL)
+    - WASI P2 defines WIT as a IDL for wasm modules components. Greatly simplifying message parsing capabilities. However, as WASIP2 is (very) new, support is quite limited (unsupported in wazero for example).
```

Comment:
```suggestion
    - WASI P2 defines WIT as a IDL for WASM modules components which greatly simplifying message parsing capabilities. However, as WASIP2 is (very) new, support is quite limited (unsupported in wazero for example).
```

Reply:
needed an additional change. Fixed in 332a9826048208293dfe9dfb96a5556539e18a97